### DISCUSSION TITLE
How to use Azure Openai Embedding Model for Knowledgebase?

### DISCUSSION THREAD

**Discussion by thunderbug1** [lobehub/lobehub#5037] [General Question]
I would like to use Lobe-Chat with a docker-compose setup using Azure Openai Models.
The Problem is that I can only configure the Chat Models but the Embedding model that is used for knowledgebases seems to be hard coded to Openai or at least I could not find documentation for how to use an Azure Openai Embedding Model.

Based on This I was able to set it up: https://lobehub.com/docs/self-hosting/server-database/docker-compose
Based on This I configured the env variables for the Azure Openai Models: https://lobehub.com/docs/self-hosting/examples/azure-openai

**Comment by thunderbug1** [comment]
This issue seems to be related: https://github.com/lobehub/lobe-chat/issues/3785

**Comment by SpeedupMaster** [comment]
Azure OpenAI Embedding Model is not supported at the moment.

**Comment by SpeedupMaster** [comment]
Perhaps you could try using the [one-api](https://github.com/songquanpeng/one-api) project to utilize the Azure OpenAI Embedding Model through model mapping. Theoretically, it should be possible, but I haven't tried it.

![image](https://github.com/user-attachments/assets/e8255e40-ce89-4a00-9632-d97d0e8b7f0b)

**Comment by beliven-daniele-sarnari** [comment]
This will help you https://github.com/lobehub/lobe-chat/pull/8075

Then you can change the following env
`DEFAULT_FILES_CONFIG=embedding_model=azure/text-embedding-3-large`
